src/averageAges.js

```diff
@@ -55,7 +69,19 @@ function calculateWomenAverageAge(people, withChildren) {
  * @return {number}
  */
 function calculateAverageAgeDiff(people, onlyWithSon) {
-  // write code here
+  const children = onlyWithSon
+    ? people.filter(p => p.sex === 'm'
+      && people.some(m => m.name === p.mother))
+    : people.filter(p => people.some(m => m.name
+      === p.mother));
+
+  let sum = 0;
+
+  children.forEach((ch) => sum += ch.born - people.find(
```

Comment:
You missed the rule: **_Arrow function should not return the assignment._**
To fix this, make the assignment not a part of the return statement.
```suggestion
  children.forEach(
    (ch) => (sum += ch.born - people.find((m) => m.name === ch.mother).born)
  );
```